packages/hash/frontend/src/pages/[account-slug]/entity-types/property-list-card.tsx

```diff
@@ -0,0 +1,701 @@
+                            color: "transparent",
+                          },
+                        },
+                    }}
+                    inputProps={{ sx: { textOverflow: "ellipsis" } }}
+                  />
+                </TableCell>
+                <TableCell>
+                  <PropertyMenu
+                    onRemove={() => {
+                      const onlyItem = created.length === 1;
+                      setCreated((list) => list.filter((item) => item !== id));
+                      if (onlyItem) {
+                        setAddingNewProperty(true);
+                        onCancel();
+                      }
+                    }}
+                  />
+                </TableCell>
+              </TableRow>
+            ))}
+          </TableBody>
+          <TableFooter>
+            {addingNewProperty ? (
+              <InsertPropertyRow
+                inputRef={sharedRef}
+                onCancel={() => {
+                  if (!created.length) {
+                    onCancel();
+                  } else {
+                    setAddingNewProperty(false);
+                  }
+                }}
+                onAdd={() => {
+                  setAddingNewProperty(false);
+                  setCreated((list) => [
+                    ...list,
+                    (Math.random() + 1).toString(36).substring(7),
+                  ]);
+                }}
+              />
+            ) : (
+              <TableRow>
+                <TableCell
+                  colSpan={
+                    // Sufficiently large to span full width
+                    100
+                  }
+                  sx={{
+                    p: "0 !important",
+                  }}
+                >
+                  <ButtonBase
+                    disableRipple
+                    disableTouchRipple
+                    onClick={() => {
+                      setAddingNewProperty(true, () => {
+                        addingNewPropertyRef.current?.focus();
+                      });
+                    }}
+                    sx={(theme) => ({
+                      color: theme.palette.gray[50],
+                      py: 1.5,
+                      width: "100%",
+                      borderRadius: 1,
+                      "&:hover": {
+                        backgroundColor: theme.palette.gray[10],
+                        color: theme.palette.gray[70],
+                      },
+                    })}
+                  >
+                    <StyledPlusCircleIcon />
+                    <Typography
+                      variant="smallTextLabels"
+                      fontWeight={500}
+                      ml={1}
+                    >
+                      Add a property
+                    </Typography>
+                  </ButtonBase>
+                </TableCell>
+              </TableRow>
+            )}
+          </TableFooter>
+        </Table>
+      </Box>
+    </WhiteCard>
+  );
+};
```

Comment:
Think the main problem here is actually that there's no logical place to split files because components are too big. I've split them up (in their own files). Let me know what you think